x-pack/plugins/canvas/canvas_plugin_src/strings/apply_strings.ts

```diff
@@ -28,3 +39,31 @@ export const applyElementStrings = (elements: ElementFactory[]) =>
 
     return () => result;
   });
+
+export const applyFunctionStrings = (functions: FunctionSpecFactory[]) =>
+  functions.map(spec => {
+    const result = spec();
+    const { name } = result;
+    const helpStrings = functionHelp[name];
+    // If we have registered strings for this spec, we should replace any that are available.
+    if (helpStrings) {
+      const { help } = helpStrings;
+      // If the function has a registered help string, replace it on the spec.
+      if (help) {
+        result.help = help;
```

Comment:
Should it be an error if the user puts a string in functionHelp *and* in the result returned from the spec?  Same below, I'm a little confused why are we replacing strings? Is one translated and one not?  Is the plan to remove the non translated strings from the function specs?

Reply:
This is a move for plugin developers.  @rashidkpc and I discussed the best way to both 1. provide i18n strings for Canvas object maintained by Elastic and 2. not disrupt the source code for people writing plugins.

I've opted to include the strings in the JS spec so developers know to supply them, (as well as honor the eventual TypeScript definition we'll apply), and then replace the string with an i18n string behind-the-scenes... this will tick multiple boxes:

- [x] Internationalize Elastic-supported components
- [x] Not "muddy" the code with a lot of i18n calls
- [x] Not confuse external developers with an Elastic-only bit of logic
- [x] Abstract away the i18n dependency
- [x] Honor a strict TypeScript definition